ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -1424,10 +1433,9 @@ public LocalResource localizeResource(
         return createLocalResource(destFS, dest, type, LocalResourceVisibility.PRIVATE);
       }
       try {
-        if (src.toUri().getScheme()!=null) {
-          FileUtil.copy(src.getFileSystem(conf), src, destFS, dest, false, false, conf);
-        }
-        else {
+        if (!isSrcLocal) {
+          FileUtil.copy(srcFs, src, destFS, dest, false, false, conf);
+        } else {
           destFS.copyFromLocalFile(false, false, src, dest);
```

Comment:
Guess we can ditch the if-else and ``isSrcLocal`` part.
``destFS.copyFromLocalFile(false, false, src, dest);``  resolves to 
``FileUtil.copy(getLocal(conf), src, this, dst, delSrc, overwrite, conf);``
which is technically same as the above `FileUtils.copy.` 
The FileSystem anyway we have resolved to Local above :
```
} else {
  srcFs = FileSystem.getLocal(conf);
}
```

Reply:
done